src/rsz/src/Resizer.tcl

```diff
@@ -125,49 +125,138 @@ proc set_layer_rc {args} {
 }
 
 sta::define_cmd_args "set_wire_rc" {[-clock] [-signal] [-data]\
-                                      [-layer layer_name]\
+                                      [-layers layers]\
+                                      [-layer layer]\
+                                      [-h_resistance h_res]\
+                                      [-h_capacitance h_cap]\
+                                      [-v_resistance v_res]\
+                                      [-v_capacitance v_cap]\
                                       [-resistance res]\
                                       [-capacitance cap]\
                                       [-corner corner]}
 
 proc set_wire_rc { args } {
   sta::parse_key_args "set_wire_rc" args \
-    keys {-layer -resistance -capacitance -corner} \
+    keys {-layer -layers -resistance -capacitance -corner \
+          -h_resistance -h_capacitance -v_resistance -v_capacitance} \
     flags {-clock -signal -data}
 
   set corner [sta::parse_corner_or_all keys]
 
-  set wire_res 0.0
-  set wire_cap 0.0
-  if { [info exists keys(-layer)] } {
-    if { [info exists keys(-resistance)] \
-           || [info exists keys(-capacitance)] } {
-      utl::error RSZ 1 "Use -layer or -resistance/-capacitance but not both."
+  set h_wire_res 0.0
+  set h_wire_cap 0.0
+  set v_wire_res 0.0
+  set v_wire_cap 0.0
+
+  if { [info exists keys(-layers)] } {
+    if { [info exists keys(-h_resistance)] \
+           || [info exists keys(-h_capacitance)] \
+           || [info exists keys(-v_resistance)] \
+           || [info exists keys(-v_capacitance)]  } {
+      utl::error RSZ 1 "Use -layers or -resistance/-capacitance but not both."
+    }
+    if { [info exists keys(-layer)] } {
+      utl::error RSZ 6 "Use -layers or -layer but not both."
     }
+    set total_h_wire_res 0.0
+    set total_h_wire_cap 0.0
+    set total_v_wire_res 0.0
+    set total_v_wire_cap 0.0
+
+    set h_layers 0
+    set v_layers 0
+
+    set layers $keys(-layers)
+
+    foreach layer_name $layers {
```

Comment:
There should be a check that exactly two layers were specified - one H and one V

Reply:
I was following the comments on the issue that if more then one layer for a direction is passed we get the average values, should I change for only receiving 2 layers?

Reply:
That's ok but we should ensure there is at least one H and one V specified.

Reply:
Ok!